apps/src/templates/projects/projectsModule.js

```diff
@@ -0,0 +1,30 @@
+/** @file Redux actions and reducer for the Projects Gallery */
+import {registerReducers} from '@cdo/apps/redux';
+
+// Actions
+const TOGGLE_GALLERY = 'Projects/TOGGLE_GALLERY';
+
+const isPublic = window.location.pathname.startsWith('/projects/public');
+
+// Reducer
+function pageLocation(state, action) {
+  const defaultState = isPublic ? 'PUBLIC' : 'PRIVATE';
```

Comment:
I would rename defaultState to initialState and move it outside of the reducer. Then you can do
```js
const initialState = isPublic ? 'PUBLIC' : 'PRIVATE";
function pageLocation(state=initialState, action) {
   // action handling here

}